R/combineFeatures.R

```diff
@@ -320,5 +318,10 @@ robust_summary <- function(e, nIter = 100, residuals = FALSE, ...) {
     ## resids <- fit$residuals
     ## se <- unique(summary(fit)$coefficients[sampleid, 'Std. Error'])
     ## was X[, sampleid, drop = FALSE] %*% fit$coefficients[sampleid]
-    return(fit$coefficients[sampleid])
+
+    ## Put NA for the samples without any expression value
+    present = apply(e,2,function(x){any(!is.na(x))})
+    out = rep(NA,length(present))
+    out[present] = fit$coefficients[sampleid]
```

Comment:
If you avoid `as.numeric` in line 288 you could reuse `p` here: `colSums(p) == nrow(e)`. The whole three lines could be simplified to `fit$coefficients[colSums(p) == nrow(e)]`.

Reply:
I don't think this will do.
I need to return  a summarised value (the sample coefficients) for every sample in the msnset.
If that sample happen to have no expression values for the given protein, this value should be `NA`.

In your suggested solution,  only values are returned for the samples without missing data and so MSnbase will probably throw an error (or should).

Reply:
Right. I somehow messed up here.
Now I would suggest:
```r
present <- rep(NA_logical_, ncol(e))
present[colSums(!is.na(e)) > 0] <- TRUE
fit$coefficients[present]
```